src/hooks/room/useRoomAvatar.ts

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2025 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Commercial
+Please see LICENSE files in the repository root for full details.
+*/
+
+import { useEffect, useState } from "react";
+import { EventType, type MatrixEvent, type Room, RoomStateEvent } from "matrix-js-sdk/src/matrix";
+
+import { useTypedEventEmitter } from "../useEventEmitter";
+
+/**
+ * Helper to retrieve the avatar for given room
+ * @returns the current avatar
+ */
+export function useRoomAvatar(room?: Room): string | undefined {
```

Comment:
Can this use `useRoomState`

Reply:
Yeah, looks like it. I've dropped this hook.